workflows/ingest/masters.go

```diff
@@ -96,14 +106,127 @@ func VBMaster(ctx workflow.Context, params VBMasterParams) (*VBMasterResult, err
 		return nil, err
 	}
 
-	if report.TopLevelInfo.Error == 0 {
+	return &MasterResult{
+		Report:  report,
+		AssetID: result.AssetID,
+		Path:    path,
+	}, nil
+}
+
+func Masters(ctx workflow.Context, params MasterParams) (*MasterResult, error) {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Starting VBMaster workflow")
+
+	ctx = workflow.WithActivityOptions(ctx, wfutils.GetDefaultActivityOptions())
+
+	result, err := uploadMaster(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	result.AnalyzeResult, err = analyzeAudioAndSetMetadata(ctx, result.AssetID, result.Path)
+
+	if result.Report.TopLevelInfo.Error == 0 {
+		err = postImportActions(ctx, []string{result.AssetID}, params.Metadata.JobProperty.Language)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
+
+func VBMaster(ctx workflow.Context, params MasterParams) (*MasterResult, error) {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Starting VBMaster workflow")
+
+	ctx = workflow.WithActivityOptions(ctx, wfutils.GetDefaultActivityOptions())
+
+	result, err := uploadMaster(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	if result.Report.TopLevelInfo.Error == 0 {
 		err = postImportActions(ctx, []string{result.AssetID}, params.Metadata.JobProperty.Language)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	return nil, nil
+	return result, nil
+}
+
+func analyzeAudioAndSetMetadata(ctx workflow.Context, assetID string, path utils.Path) (*common.AnalyzeEBUR128Result, error) {
+	var result common.AnalyzeEBUR128Result
+	err := wfutils.ExecuteWithQueue(ctx, activities.AnalyzeEBUR128Activity, activities.AnalyzeEBUR128Params{
+		FilePath:       path.WorkerPath(),
+		TargetLoudness: 0.0, // TODO: Determine our target loudness
+	}).Get(ctx, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	values := map[string]float64{
+		// TODO: determine the correct values for these fields
+		vscommon.FieldLoudnessLUFS.Value:  result.IntegratedLoudness,
+		vscommon.FieldTruePeak.Value:      result.TruePeak,
+		vscommon.FieldLoudnessRange.Value: result.LoudnessRange,
+	}
+
+	keys, err := wfutils.GetMapKeysSafely(ctx, values)
+	if err != nil {
+		return nil, err
+	}
+	for _, key := range keys {
+		err = wfutils.SetVidispineMeta(ctx, assetID, key, strconv.FormatFloat(values[key], 'f', 2, 64))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &result, nil
+}
```

Comment:
@KillerX Can you take a look at this?

Reply:
Put -24 as the target. The analyze part in general does not really care about the target, it's only used to calc suggested adjustment. The Values in 171+ seem correct.